src/app/exam/modals/AddProblemModal.js

```diff
@@ -1,87 +1,55 @@
-import './AddParticipantModal.scss'
+import './AddProblemModal.scss'
 import React, { createRef, useState } from "react";
 import { renderModal } from "../../commons/modals/modal";
 import { ModalHeader } from "../../commons/modals/ModalHeader";
 
-const outsideDivStyle = {
-    padding: "20px 40px",
-};
-
-const inputGroupStyle = {
-    display: "flex",
-    flexDirection: "row",
-    justifyContent: "space-between",
-
-};
-
-const labelStyle = {
-    margin: "5px 10px",
-    fontSize: "20px",
-    fontWeight: "500",
-    lineHeight: "30px",
-};
-
-const inputStyle = {
-    margin: "5px 10px",
-    flexGrow: "1",
-    backgroundColor: "#F3F3F3",
-    border: "1px solid #A2A3B1",
-    boxSizing: "border-box",
-    borderRadius: "10px",
-};
-
-const btnStyle = {
-    background: "#91CB46",
-    color: "#FFF",
-    borderRadius: "10px",
-    padding: "7px 10px",
-
-    fontSize: "18px",
-    lineHeight: "27px",
-};
-
 const InputField = ({ id, type, labelText, value, placeholder, onChange }) => (
-    <div style={inputGroupStyle}>
-        <label htmlFor={id} style={labelStyle}>{labelText}</label>
-        <input id={id} type={type} onChange={onChange} value={value} placeholder={placeholder} style={inputStyle} />
+    <div className="input-field">
+        <label htmlFor={id}>{labelText}</label>
+        <input id={id} type={type} onChange={onChange} value={value} placeholder={placeholder} required={true} />
     </div>
 );
 
 const AddProblemModal = ({ title, show, onClose, onSubmit }) => {
-    const closeIconRef = createRef();
+    const closeIconRef = createRef(), formRef = createRef();
     const [problemId, setProblemId] = useState('');
     const [scorePercentage, setScorePercentage] = useState('');
     const [submissionQuota, setSubmissionQuota] = useState('');
 
-    const submit = () => {
-        if (!problemId || !scorePercentage || !submissionQuota) return;
-
-        onSubmit(problemId, scorePercentage, submissionQuota);
-
+    const clearFields = () => {
         setProblemId('');
         setScorePercentage('');
         setSubmissionQuota('');
-    };
+    }
+
+    const handleFormSubmit = e => {
+        e.preventDefault();
+        const form = formRef.current;
 
-    const enterSubmit = e => {
-        if (e.keyCode === 13) submit();
+        if (form.checkValidity()) {
+            onSubmit(problemId, scorePercentage, submissionQuota).then(clearFields);
```

Comment:
you pass `problemId, scorePercentage, submissionQuota` to `onSubmit`
which is weak-typed, thus **evil**, please don't do this otherwise we'd better bring TypeScript back.

Please:
(1) rename `onSubmit` to `onSubmitQuestion` 
(2) please create a `Question` and pass the question into the `onSubmitQuestion` method.

in such way, we can embrace the dynamic typing system without compromising the maintainability.

Reply:
I'm not sure if renaming `onSubmit` to `onSubmitQuestion` is a good rename.
This `onSubmit` means to tell this modal when this form needs to submit something, pass the arguments into this function.

Reply:
Rename done.